component/pprof/scrape/scrape.go

```diff
@@ -0,0 +1,310 @@
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/agent/component/pprof"
+	"github.com/grafana/agent/component/prometheus/scrape"
+)
+
+const (
+	pprofMemory     string = "memory"
+	pprofBlock      string = "block"
+	pprofGoroutine  string = "goroutine"
+	pprofMutex      string = "mutex"
+	pprofProcessCPU string = "process_cpu"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name: "pprof.scrape",
+		Args: Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments holds values which are used to configure the pprof.scrape
+// component.
+type Arguments struct {
+	Targets   []discovery.Target `river:"targets,attr"`
+	ForwardTo []pprof.Appendable `river:"forward_to,attr"`
+
+	// The job name to override the job label with.
+	JobName string `river:"job_name,attr,optional"`
+	// A set of query parameters with which the target is scraped.
+	Params url.Values `river:"params,attr,optional"`
+	// How frequently to scrape the targets of this scrape config.
+	ScrapeInterval time.Duration `river:"scrape_interval,attr,optional"`
+	// The timeout for scraping targets of this config.
+	ScrapeTimeout time.Duration `river:"scrape_timeout,attr,optional"`
+	// The URL scheme with which to fetch metrics from targets.
+	Scheme string `river:"scheme,attr,optional"`
+
+	// todo(ctovena): add support for limits.
+	// // An uncompressed response body larger than this many bytes will cause the
+	// // scrape to fail. 0 means no limit.
+	// BodySizeLimit units.Base2Bytes `river:"body_size_limit,attr,optional"`
+	// // More than this many targets after the target relabeling will cause the
+	// // scrapes to fail.
+	// TargetLimit uint `river:"target_limit,attr,optional"`
+	// // More than this many labels post metric-relabeling will cause the scrape
+	// // to fail.
+	// LabelLimit uint `river:"label_limit,attr,optional"`
+	// // More than this label name length post metric-relabeling will cause the
+	// // scrape to fail.
+	// LabelNameLengthLimit uint `river:"label_name_length_limit,attr,optional"`
+	// // More than this label value length post metric-relabeling will cause the
+	// // scrape to fail.
+	// LabelValueLengthLimit uint `river:"label_value_length_limit,attr,optional"`
+
+	HTTPClientConfig component_config.HTTPClientConfig `river:"http_client_config,block,optional"`
+
+	ProfilingConfig *ProfilingConfig `river:"profiling_config,block,optional"`
+}
+
+type ProfilingConfig struct {
+	PprofConfig PprofConfig `river:"pprof_config,attr,optional"`
+	PprofPrefix string      `river:"path_prefix,attr,optional"`
+}
+
+type PprofConfig map[string]*PprofProfilingConfig
+
+type PprofProfilingConfig struct {
+	Enabled *bool  `river:"enabled,attr,optional"`
+	Path    string `river:"path,attr,optional"`
+	Delta   bool   `river:"delta,attr,optional"`
+}
+
```

Comment:
I don't know about making this a map. Labeled blocks might be more appropriate so you have

```river 
profile.memory {
  path = "/debug/pprof/allocs" 
}

profile.block {
  path = "/debug/pprof/block" 
}

...

profile.custom "label" {
  path = ... 
}
```

Using different block names for the recognized profile names allows us to retain the defaults, but we can still support custom ones with the `profile.custom` block. 

Also: I'm not sure why you'd want to pass `enabled` here to the individual profile configs.

Reply:
We discussed this offline and we're going to try to do these as blocks, keeping in mind the need for defaults for known profile endpoints and the ability to add custom profile endpoints. 

We will be conscious about the ability to convert it into a map so it fits well with the existing code.

Reply:
Sounds good, let me know if I can help.

There are a couple of interesting issues related to this already in Phlare, but the most noteworthy https://github.com/grafana/phlare/issues/456. This configuration varies from one application to another.

We can also consider fgprof not custom but disable by default. Having the ability to add custom profile types is nice to have.

I suspect in the future we will add more per profile configuration, for example, to exclude some sample types (just send sample value, not sample count).

Reply:
Done:

```river 
phlare.scrape "local" {
  targets    = [
    {"__address__" = "localhost:4100", "app"="phlare"},
    {"__address__" = "localhost:12345", "app"="agent"},
  ]
  forward_to = [phlare.write.local.receiver]
  profiling_config {
    profile.fgprof {
      enabled = true
    }
    profile.block {
      enabled = false
    }
    profile.mutex {
      enabled = false
    }
  }
}
```